xdsl/parser/attribute_parser.py

```diff
@@ -988,6 +988,120 @@ def to_type(self, parser: AttrParser, type: AnyFloat | IntegerType | IndexType):
                         parser, allow_negative=True, allow_booleans=False
                     )
 
+    def _parse_optional_int(self) -> tuple[int, Span] | None:
+        """Parses an optional int. Includes parsing the minus token.
```

Comment:
We usually start the docs on newlines, this is more of a nit, but feels like it would be good to stay consistent:
```suggestion
        """
        Parses an optional int. Includes parsing the minus token.
```

Reply:
This function was removed [here](https://github.com/xdslproject/xdsl/pull/4351/commits/da4ad24a9c36572a59f83050eb0eab973312bb75)